libenv/sysinfo.c

```diff
@@ -1645,6 +1659,88 @@ static void OSClasses(EvalContext *ctx)
 
 /*********************************************************************************/
 
+#ifdef __APPLE__
+static void Apple_Version(EvalContext *ctx)
+{
+    FILE *pp = NULL;
+
+    if ((!FileCanOpen("/usr/bin/sw_vers", "r") || ((pp = cf_popen("/usr/bin/sw_vers", "r", true)) == NULL)))
+    {
+        Log(LOG_LEVEL_VERBOSE, "Could not find macOS system version information.");
+        return;
+    }
+
+    size_t line_size = CF_BUFSIZE;
+    char *line = xmalloc(line_size), *r;
+    int revcomps = 0, major, minor, patch;
+    char *flavor = NULL;
+
+    while (CfReadLine(&line, &line_size, pp) != -1)
+    {
+        if ((r = strstr(line, "ProductName")))
+        {
+            r = strrchr(line, '\t');
+            if (!r || !++r)
+            {
+                continue;
+            }
+            for (size_t i = 0; i < strlen(r); i++)
+            {
+                r[i] = tolower(r[i]);
+            }
+            EvalContextClassPutHard(
+                    ctx,
+                    r,
+                    "inventory,attribute_name=none,source=agent,derived-from=sw_vers");
+            flavor = SafeStringDuplicate(r);
+        }
+        else if ((r = strstr(line, "ProductVersion")))
+        {
+            r = strrchr(line, '\t');
+            if (!r || !++r)
+            {
+                continue;
+            }
+            revcomps = sscanf(r, "%d.%d.%d", &major, &minor, &patch);
```

Comment:
Consider using `unsigned int` variable and `%u` format specifier to make sure we never get a negative version number
```suggestion
            revcomps = sscanf(r, "%u.%u.%u", &major, &minor, &patch);
```

Reply:
✅